src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -14,29 +16,140 @@ namespace PInvoke
     /// </content>
     public static partial class AdvApi32
     {
-        public static void InstallService(string servicePath, string serviceName, string serviceDisplayName, string serviceDescription, string userName, string password)
+        /// <summary>
+        /// Enumerates services in the specified service control manager database.
+        /// The name and status of each service are provided.
+        /// </summary>
+        /// <returns>
+        /// An IEnumerable of <see cref="ENUM_SERVICE_STATUS"/> structures that receive the name and service status information for each service in the database.
+        /// </returns>
+        /// <exception cref="Win32Exception">If the method fails, returning the calling thread's last-error code value.</exception>
+        public static IEnumerable<ENUM_SERVICE_STATUS> EnumServicesStatus()
+        {
+            using (var scmHandle = OpenSCManager(null, null, ServiceManagerAccess.SC_MANAGER_ENUMERATE_SERVICE))
+            {
+                if (scmHandle.IsInvalid)
+                {
+                    throw new Win32Exception();
+                }
+
+                int buffferSizeNeeded = 0;
+                int numServicesReturned = 0;
+                int resumeIndex = 0;
+                if (!EnumServicesStatus(
+                    scmHandle,
+                    ServiceType.SERVICE_WIN32,
+                    ServiceStateQuery.SERVICE_STATE_ALL,
+                    IntPtr.Zero,
+                    0,
+                    ref buffferSizeNeeded,
+                    ref numServicesReturned,
+                    ref resumeIndex))
+                {
+                    var lastError = GetLastError();
+                    if (lastError != Win32ErrorCode.ERROR_MORE_DATA)
+                    {
+                        throw new Win32Exception(lastError);
+                    }
+
+                    IntPtr buffer = Marshal.AllocHGlobal(buffferSizeNeeded);
+                    if (!EnumServicesStatus(
+                        scmHandle,
+                        ServiceType.SERVICE_WIN32,
+                        ServiceStateQuery.SERVICE_STATE_ALL,
+                        buffer,
+                        buffferSizeNeeded,
+                        ref buffferSizeNeeded,
+                        ref numServicesReturned,
+                        ref resumeIndex))
+                    {
+                        throw new Win32Exception();
+                    }
+
+                    int bufferSize = buffer.ToInt32();
```

Comment:
It's named `bufferSize` but it look like it's used as a current possition in the buffer